### ISSUE TITLE
Unable to download Xcode with Federated Apple ID

### ISSUE THREAD

**Issue by alteral** [RobotsAndPencils/XcodesApp#346]
**Describe the bug**
- Unable to download Xcode without authorization
<img width="596" alt="Screen Shot 2023-01-03 at 7 47 36 PM" src="https://user-images.githubusercontent.com/20794991/210430490-7bb92252-9808-44bb-be49-bb23e2dfee93.png">

- Unable to log in with Federated Apple ID
<img width="592" alt="Screen Shot 2023-01-03 at 7 48 26 PM" src="https://user-images.githubusercontent.com/20794991/210430500-237a8462-0e36-42f3-baa1-1f4a02920f4b.png">

**Expected behavior**
- Authorization can be done through [Federated Authentication](https://support.apple.com/en-gb/guide/apple-business-manager/axmb19317543/1/web/1)

or

- Xcode can be downloaded without authorization

**Version**
 - macOS: 12.5.1 (21G83)
 - Xcodes: 1.9.0

**Comment by dazigna** [comment]
Currently facing the same issue. With a business-managed account we are not able to download new xcodes version since we can't authenticate.
Any idea how to solve that ?
thanks!

**Comment by MattKiazyk** [comment]
@dazigna @alteral Hi!

Currently we don't support logging in through Federated Authentication. I don't have any way to see how it works, as I'm not part of any federated organization. 

Can you can easily create your own personal apple id instead?

**Comment by dazigna** [comment]
Hi @MattKiazyk Thanks for the quick answer, This is what I ended up doing. Federated Authentication is some kind of Oauth but tbf I do not know how this works under the hood.

**Comment by alteral** [comment]
Hi @MattKiazyk, thanks for your suggestion, this is what I ended up doing too.

Isn't it possible to allow downloading xcodes without authorisation?

**Comment by MattKiazyk** [comment]
@alteral that is correct at the moment. There was a brief moment this past year where we could get a token that allowed downloading without logging in, however Apple returns 403 forbidden now on those calls. 

I would love to have a way to not log in, or at least use an App-Specific password, but I haven't figured out how yet.

**Comment by alteral** [comment]
Gotcha. Ok, thanks 👍

**Comment by Mcrich23** [comment]
You can also download using http://xcodereleases.com instead of apple via the XcodesApp

**Comment by davidpasztor** [comment]
@Mcrich23 I'm getting the same error message that tells me I need to log in with my Apple ID regardless of whether I have xcodereleases.com or Apple selected as my Downloads data source, so unfortunately that doesn't seem to resolve the Federated Apple ID issue.

**Comment by radloffl** [comment]
@MattKiazyk is this something you'd be open to contribution for? If so, seems like we should move this issue and I can spin up a PR in the new XcodesLoginKit project? 

This issue is my main hesitation for spreading the word about Xcodes at my company with 350+ Apple platform developers, our legal and tech departments have approved accepting EULA/privacy policy and install sign offs but they won't love folks using personal Apple Accounts on company owned hardware.

Apple seems to be pushing for federation more (I believe this was the reason behind the recent developments in being able to migrate Apple Account purchases). One of the major pain points of federating when my employer moved was that anyone that had used their work email as their Apple Account just lost their account and purchases when your organization decided to federate. With a more palatable off ramp for those folks now, I suspect it will become more common for businesses to move to federated accounts and this to be a blocker for suggesting using Xcodes since it'd require using a personal Apple Account.

**Comment by gabriellet** [comment]
I'm not allowed to use a personal Apple Account for work, so that workaround isn't an option for me. Would love to see this addressed, because I prefer using XCodes to manage my installations.
